content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,861 @@
+
+
+We can use these approaches to test if the disease is linked to any of the other
+recorded variables. As illness itself is a categorical variable, only approaches
+of type 2 and 3 will be illustrated here.
+
+
+### Is illness linked to age?
+
+We can use the function `t.test()` to test if the average age is different across
+illness status. As this test assumes that the two categories exhibit similar
+variation, we first ensure that the variances are comparable using Bartlett's
+test. The syntax `variable ~ group` is used to indicate the variable of interest (left hand-side), and the group
+ (right hand-side):
+
+```{r stegen_bartlett_test}
+bartlett.test(stegen$age ~ stegen$ill)
+```
+
+The resulting p-value (`r round(bartlett.test(stegen$age ~ stegen$ill)[["p.value"]], 3)`) suggests the variances are indeed comparable. Wecan thus proceed to a $t$-test:
+
+
+```{r stegen_t_test}
+t.test(stegen$age ~ stegen$ill, var.equal = TRUE)
+```
+
+The p-value of `r round(t.test(stegen$age ~ stegen$ill, var.equal = TRUE)[["p.value"]], 3)` confirms the previous graphics: the illness does not seem to be linked to age.
+
+
+
+### Is illness linked to gender?
+
+To test the association between gender and illness (2 categorical variables), we
+first build a 2-by-2 (contingency) table, using:
+
+```{r stegen_2by2}
+tab_ill_sex <- table(stegen$ill, stegen$sex)
+tab_ill_sex
+```
+
+Note that proportions can be obtained using `prop.table`:
+
+```{r stegen_prop_table}
+## basic proportions
+prop.table(tab_ill_sex)
+
+## expressed in %, rounded:
+round(100 * prop.table(tab_ill_sex))
+```
+
+Once a contingency table has been built, the Chi-square test can be run using `chisq.test`:
+
+```{r stegen_chisq}
+chisq.test(tab_ill_sex)
+```
+
+Here, the p-value of `r round(chisq.test(tab_ill_sex)[["p.value"]], 3)` suggests illness is not related to sex either.
+
+
+## Making multiple tests
+
+The natural next step is to run multiple tests for all potential risk factors
+recorded. There are many ways to go about this in **R**, but we will illustrate
+one of the most common workflows:
+
+1. isolate a subset of variables to test (the ones indicative of food consumption) (using `[]`)
+2. for each:
+    i) build contingency table with illness status (using `table`)
+    ii) run a Chi-square test on this table (using `chisq.test`)
+
+All of these have been seen so far; the only missing piece is step 2), which we
+will cover using a very handy function called `lapply()`.
+
+
+### Isolating the variables to test
+
+We use the natural structure of the dataset, as variables representing food
+consumption are all the last columns:
+
+```{r stegen_isolate_food}
+names(stegen)
+```
+
+In this case, we need to retain columns 6 to 21, excluding `tportion` and
+`mportion`, which are not binary; this can be done using the subsetting operator
+`[...]`, where `...` are numbers indicating the columns to retain:
+
+```{r stegen_food}
+to_keep <- c(6, 8, 9, 10, 12:21)
```

Comment:
maybe mention that you could also do this using strings of var names?

Reply:
I've changed this so that it is strings of var names